packages/Telephony/Actions/SerialNumberInput.cs

```diff
@@ -113,23 +131,27 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
         public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext dc, CancellationToken cancellationToken = default)
         {
             //Check if we were interrupted. If we were, follow our logic for when we get interrupted.
-            bool wasInterrupted = dc.State.GetValue(TurnPath.Interrupted, () => false);
+            var wasInterrupted = dc.State.GetValue(TurnPath.Interrupted, () => false);
             if (wasInterrupted)
             {
                 return await PromptUserAsync(dc, cancellationToken).ConfigureAwait(false);
             }
 
             //Get value of termination string from expression
-            string regexPattern = this.RegexPattern?.GetValue(dc.State);
-            SerialNumberPattern snp = new SerialNumberPattern(regexPattern, true);
+            var batchLength = BatchLength.GetValue(dc.State);
+            var acceptAlphabet = AcceptAlphabet.GetValue(dc.State);
+            var acceptNumbers = AcceptNumbers.GetValue(dc.State);
 
-            string[] choices = dc.State.GetValue<string[]>("this.ambiguousChoices");
-            bool isAmbiguousPrompt = choices != null && choices.Length >= 2;
+            // TODO: Delete placeholder regex pattern once we remove SerialNumberPattern.
+            var regexPattern = $"([{(acceptAlphabet ? "a-zA-Z" : string.Empty)}{(acceptNumbers ? "0-9" : string.Empty)}]{{{batchLength}}})";
+            var snp = new SerialNumberPattern(regexPattern, true);
+            var choices = dc.State.GetValue<string[]>("this.ambiguousChoices");
```

Comment:
>choices [](http://example.com/codeflow?start=16&length=7)

Are you sure there are at least 2 of these always? You do an array accessor without checking for null or length...just just want to make sure that's ok.

Reply:
Ah I see what happened here, I took away the explicit typing so it looks like a new line, the functionality on the ambiguous checking was otherwise unchanged.

From looking at the code and test case, when we reach this part of the flow, the choices are at least always two because the end user is choosing between one option or another, e.g., "Did you mean A or 8? Press 1 for A and 2 for 8"